src/systemctl/systemctl.c

```diff
@@ -8822,16 +8821,20 @@ static int reload_with_fallback(void) {
 }
 
 static int start_with_fallback(void) {
+        int r;
+
         /* First, try systemd via D-Bus. */
-        if (start_unit(0, NULL, NULL) == 0)
+        r = start_unit(0, NULL, NULL);
+        if (r >= 0)
```

Comment:
start_unit() returns positive errors (because it returns sysv compatible process exit codes). THis doesn't look right hence.

Reply:
Ah, right. I removed most of that patch now, just the `#ifdef` bits remain.